lib/autoupdate/agent/updater.go

```diff
@@ -341,18 +352,50 @@ func (u *Updater) Install(ctx context.Context, override OverrideConfig) error {
 	// Only write the configuration file if the initial update succeeds.
 	// Note: skip_version is never set on failed enable, only failed update.
 
-	if err := writeConfig(u.ConfigPath, cfg); err != nil {
+	if err := writeConfig(u.UpdateConfigPath, cfg); err != nil {
 		return trace.Wrap(err, "failed to write %s", updateConfigName)
 	}
 	u.Log.InfoContext(ctx, "Configuration updated.")
 	return trace.Wrap(u.notices(ctx))
 }
 
+// findAgentProxy finds a proxy in teleport.yaml if not specified or set in update configuration.
+func (u *Updater) findAgentProxy(ctx context.Context) string {
+	fc, err := config.ReadFromFile(u.TeleportConfigPath)
+	if err != nil {
+		u.Log.DebugContext(ctx, "Unable to find proxy in Teleport config", "config", u.TeleportConfigPath, errorKey, err)
+		return ""
+	}
+	addr, err := libutils.ParseHostPortAddr(fc.ProxyServer, libdefaults.HTTPListenPort)
+	if err != nil {
+		u.Log.DebugContext(ctx, "Unable to parse proxy in Teleport config", "config", u.TeleportConfigPath, errorKey, err)
+		return ""
+	}
+	return addr.String()
+}
+
+// sameProxies returns true if both proxies addresses are the same.
+func sameProxies(a, b string) bool {
+	if a == b {
+		return true
+	}
+	addrA, err := libutils.ParseAddr(a)
+	if err != nil {
+		return false
+	}
+	addrB, err := libutils.ParseAddr(b)
+	if err != nil {
+		return false
+	}
+	return addrA.Host() == addrB.Host() &&
+		addrA.Port(443) == addrB.Port(443)
```

Comment:
in the teleport.yaml i believe the default port for the proxy address, if it's not specified, is 3080. Does the updater default to 443? If so, we probably don't want to compare two addresses with unset ports as equal

Reply:
nit: we might same constant defined here

https://github.com/gravitational/teleport/blob/f14c4c75aa081a7deaefb0d4f95456dbbd338b45/lib/web/apiserver.go#L5020-L5025

Reply:
@nklaassen You are correct that the updater defaults to 443 (no port), while `teleport.yaml` defaults to 3080. However, `findProxyAgent` converts the `teleport.yaml`-format into `[host]:[port]` format, defaulting empty port to 3080:
https://github.com/gravitational/teleport/pull/51633/files#diff-cf29b17bebb0e24bf48bacda13ae9ab300a3c718ccf38a97ea0cccbb2d5c11dfR369

This ensures we compare similar addresses. Both `a` and `b` are expected to be in the same format.

Reply:
Added clarifying comments.